client/client.go

```diff
@@ -18,6 +18,8 @@ type Client interface {
 	BulkIndexClose(context.Context) error
 	Checker(ctx context.Context, state *health.CheckState) error
 	CreateIndex(ctx context.Context, indexName string, indexSettings []byte) error
+	DeleteDocumentByID(ctx context.Context, index, documentID string) error
+	DeleteDocumentByURI(ctx context.Context, index, uri string) error
```

Comment:
The ES client should not have implementation specific logic. `uri` is a specific field used in documents added by the search api/importer etc. This library is potentially usable by other unrelated apps therefore all logic should be generic wrappers around core ES functionality. Instead, the client should wrap the `_delete_by_query` functionality and then the app using it can supply the implementation specific functionality (the query containing uri in this case).
```suggestion
	DeleteDocumentsBySearch(ctx context.Context, search client.Search) error
```

Reply:
NB. I've tried to keep the 'search' terminology and parameters similar to the `Search` function for consistency, even though I'm not entirely comfortable with the precedent. Happy to defer to Linden if he disagrees.